tests/ignite/metrics/test_root_mean_squared_error.py

```diff
@@ -17,24 +18,52 @@ def test_zero_div():
 
 
 def test_compute():
+
     rmse = RootMeanSquaredError()
 
-    y_pred = torch.Tensor([[2.0], [-2.0]])
-    y = torch.zeros(2)
-    rmse.update((y_pred, y))
-    assert isinstance(rmse.compute(), float)
-    assert rmse.compute() == 2.0
+    def _test(y_pred, y, batch_size):
+        rmse.reset()
+        rmse.update((y_pred, y))
+
+        np_y = y.numpy().ravel()
+        np_y_pred = y_pred.numpy().ravel()
+
+        if batch_size > 1:
+            n_iters = y.shape[0] // batch_size + 1
+            for i in range(n_iters):
+                idx = i * batch_size
+                rmse.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
+
+        np_res = np.sqrt(np.power((np_y - np_y_pred), 2.0).sum() / np_y.shape[0])
+        res = rmse.compute()
+
+        assert isinstance(res, float)
+        assert pytest.approx(res) == np_res
+
+    def get_test_cases():
+
+        test_cases = [
+            (torch.randint(0, 10, size=(100,)), torch.randint(0, 10, size=(100,)), 1),
+            (torch.randint(-10, 10, size=(100, 1)), torch.randint(-10, 10, size=(100, 1)), 1),
+            (torch.randint(0, 10, size=(200,)), torch.randint(0, 10, size=(200,)), 1),
```

Comment:
Is this test (and the following one) really relevant regarding to the test `size=(100,)` ?

Reply:
Can you please explain `relevant`? you mean `size=100` and `size=200` are the same?

Reply:
I mean `size=200` does add better test coverage regarding to `size=100`. Why do we test this size ?

Reply:
Basically we are trying multiple test sizes `(100, 200)` and multiple ranges `([0, 10] and [-10, 10])`, what do you think?

Reply:
@KickItLikeShika the question is why 100 and not 10 ? Why 4 test cases and not just 2 ?

Reply:
There is not particular reason for using 100 instead of 10. and we're trying multiple sizes `4 tests cases not two` for a better coverage.
> We already talked about that in one of your previous PR and the idea is too save computational resources.
Yes we did, but computational load is not a problem here, what do you think?

Reply:
> There is no particular reason for using 100 instead of 10. and we're trying multiple sizes `4 tests cases not two` for a better coverage.

The coverage is not better IMO. The methods don't depend on values neither shapes.

Reply:
@sdesrozis thanks for the explanation!